<font size="5.5">Analysis of WHO Life Expectancy Dataset</font>

This is an analysis of data published by the Global Health Observatory (GHO) data repository under World Health Organization (WHO). The goal of this analysis is to find if correlation exists between the life expectancy of a country and various factors such as vaccinations, Alcohol consumption, Expendeture on Healthcare etc. 

In [3]:
#importing all the needed modules
import pandas as pd
import os

In [10]:
df=pd.read_csv("Life Expectancy Data.csv")
df.head()

Country  Year      Status  Life expectancy   Adult Mortality  \
0  Afghanistan  2015  Developing              65.0            263.0   
1  Afghanistan  2014  Developing              59.9            271.0   
2  Afghanistan  2013  Developing              59.9            268.0   
3  Afghanistan  2012  Developing              59.5            272.0   
4  Afghanistan  2011  Developing              59.2            275.0   

   infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   ...  \
0             62     0.01               71.279624         65.0      1154  ...   
1             64     0.01               73.523582         62.0       492  ...   
2             66     0.01               73.219243         64.0       430  ...   
3             69     0.01               78.184215         67.0      2787  ...   
4             71     0.01                7.097109         68.0      3013  ...   

   Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  Population  \
0    6.0               8.16         65.0        0.1  584.259210  33736494.0   
1   58.0               8.18         62.0        0.1  612.696514    327582.0   
2   62.0               8.13         64.0        0.1  631.744976  31731688.0   
3   67.0               8.52         67.0        0.1  669.959000   3696958.0   
4   68.0               7.87         68.0        0.1   63.537231   2978599.0   

    thinness  1-19 years   thinness 5-9 years  \
0                   17.2                 17.3   
1                   17.5                 17.5   
2                   17.7                 17.7   
3                   17.9                 18.0   
4                   18.2                 18.2   

   Income composition of resources  Schooling  
0                            0.479       10.1  
1                            0.476       10.0  
2                            0.470        9.9  
3                            0.463        9.8  
4                            0.454        9.5  

[5 rows x 22 columns]

In [12]:
df.describe()

Year  Life expectancy   Adult Mortality  infant deaths  \
count  2938.000000       2928.000000      2928.000000    2938.000000   
mean   2007.518720         69.224932       164.796448      30.303948   
std       4.613841          9.523867       124.292079     117.926501   
min    2000.000000         36.300000         1.000000       0.000000   
25%    2004.000000         63.100000        74.000000       0.000000   
50%    2008.000000         72.100000       144.000000       3.000000   
75%    2012.000000         75.700000       228.000000      22.000000   
max    2015.000000         89.000000       723.000000    1800.000000   

           Alcohol  percentage expenditure  Hepatitis B       Measles   \
count  2744.000000             2938.000000  2385.000000    2938.000000   
mean      4.602861              738.251295    80.940461    2419.592240   
std       4.052413             1987.914858    25.070016   11467.272489   
min       0.010000                0.000000     1.000000       0.000000   
25%       0.877500                4.685343    77.000000       0.000000   
50%       3.755000               64.912906    92.000000      17.000000   
75%       7.702500              441.534144    97.000000     360.250000   
max      17.870000            19479.911610    99.000000  212183.000000   

              BMI   under-five deaths         Polio  Total expenditure  \
count  2904.000000         2938.000000  2919.000000         2712.00000   
mean     38.321247           42.035739    82.550188            5.93819   
std      20.044034          160.445548    23.428046            2.49832   
min       1.000000            0.000000     3.000000            0.37000   
25%      19.300000            0.000000    78.000000            4.26000   
50%      43.500000            4.000000    93.000000            5.75500   
75%      56.200000           28.000000    97.000000            7.49250   
max      87.300000         2500.000000    99.000000           17.60000   

       Diphtheria      HIV/AIDS            GDP    Population  \
count  2919.000000  2938.000000    2490.000000  2.286000e+03   
mean     82.324084     1.742103    7483.158469  1.275338e+07   
std      23.716912     5.077785   14270.169342  6.101210e+07   
min       2.000000     0.100000       1.681350  3.400000e+01   
25%      78.000000     0.100000     463.935626  1.957932e+05   
50%      93.000000     0.100000    1766.947595  1.386542e+06   
75%      97.000000     0.800000    5910.806335  7.420359e+06   
max      99.000000    50.600000  119172.741800  1.293859e+09   

        thinness  1-19 years   thinness 5-9 years  \
count            2904.000000          2904.000000   
mean                4.839704             4.870317   
std                 4.420195             4.508882   
min                 0.100000             0.100000   
25%                 1.600000             1.500000   
50%                 3.300000             3.300000   
75%                 7.200000             7.200000   
max                27.700000            28.600000   

       Income composition of resources    Schooling  
count                      2771.000000  2775.000000  
mean                          0.627551    11.992793  
std                           0.210904     3.358920  
min                           0.000000     0.000000  
25%                           0.493000    10.100000  
50%                           0.677000    12.300000  
75%                           0.779000    14.300000  
max                           0.948000    20.700000

Here we see that the mean life expectancy from 2000 to 2015 is 69.224932. Now, we shall analyse the mean and median life expectancy year by year from 2000 to 2015. 

In [17]:
df1=df.copy()

In [39]:
df2=df1.groupby(by="Year").mean()
df2=df2["Life expectancy "]
df2

Year
2000    66.750273
2001    67.128962
2002    67.351366
2003    67.433333
2004    67.646448
2005    68.209290
2006    68.667760
2007    69.036066
2008    69.427869
2009    69.938251
2010    70.048634
2011    70.654098
2012    70.916940
2013    71.236066
2014    71.536612
2015    71.616940
Name: Life expectancy , dtype: float64

In [40]:
df2=df1.groupby(by="Year").median()
df2=df2["Life expectancy "]
df2

Year
2000    71.0
2001    71.2
2002    71.4
2003    71.1
2004    71.2
2005    71.6
2006    72.1
2007    72.4
2008    72.4
2009    72.6
2010    72.8
2011    73.3
2012    73.2
2013    73.2
2014    73.6
2015    73.9
Name: Life expectancy , dtype: float64

As we can see life expectancy, on average, has risen in the period of 2000 and 2015. The mean life expectancy has risen from 66.75 Years to 71.61 Years, which is a rise of About 7%. The median life expectancy has increased from 71 to 73.9, which is an increase of 4%

Now we shall perform analysis of other factors like adult mortality and infant deaths

In [43]:
df2=df1.groupby(by="Year").mean()
df2=df2[["Adult Mortality","infant deaths"]]
df2

Adult Mortality  infant deaths
Year                                
2000       181.475410      37.540984
2001       175.169399      36.825137
2002       171.437158      35.584699
2003       169.131148      34.874317
2004       186.262295      33.644809
2005       167.633880      32.366120
2006       170.404372      31.808743
2007       159.808743      30.612022
2008       174.519126      29.568306
2009       163.393443      28.956284
2010       161.863388      27.928962
2011       156.535519      26.830601
2012       148.792350      26.338798
2013       148.765027      23.968912
2014       148.688525      24.557377
2015       152.863388      23.803279

As we can see the number of infant deaths(per 1000) as well as the Adult Mortality has decreased drastically

Now let us analyse the life expectancy for developed and developing countries seperately

In [49]:
df2=df[df["Status"]=="Developing"]
df2=df2.groupby(by="Year").mean()
df2[["Life expectancy ","Adult Mortality","infant deaths"]]

Life expectancy   Adult Mortality  infant deaths
Year                                                  
2000         64.619868       200.496689      45.119205
2001         65.009934       193.158940      44.258278
2002         65.190728       189.443709      42.761589
2003         65.206623       187.609272      41.913907
2004         65.370861       207.834437      40.423841
2005         66.009272       184.172185      38.887417
2006         66.450331       186.695364      38.211921
2007         66.860927       177.589404      36.768212
2008         67.413907       195.172185      35.516556
2009         67.894040       181.754967      34.788079
2010         67.908609       180.370861      33.562914
2011         68.523841       175.788079      32.231788
2012         68.898013       165.258278      31.649007
2013         69.234437       166.165563      28.484472
2014         69.501987       164.430464      29.496689
2015         69.690066       169.390728      28.596026

As we see, for developing countried, there is a significant increase in the life expectancy and a drastic decrease in adult mortality and the number of infant deaths(per 1000). 

In [50]:
df2=df[df["Status"]=="Developed"]
df2=df2.groupby(by="Year").mean()
df2[["Life expectancy ","Adult Mortality","infant deaths"]]

Life expectancy   Adult Mortality  infant deaths
Year                                                  
2000         76.803125         91.71875        1.78125
2001         77.128125         90.28125        1.75000
2002         77.546875         86.46875        1.71875
2003         77.940625         81.93750        1.65625
2004         78.384375         84.46875        1.65625
2005         78.590625         89.59375        1.59375
2006         79.131250         93.53125        1.59375
2007         79.300000         75.90625        1.56250
2008         78.931250         77.06250        1.50000
2009         79.584375         76.75000        1.43750
2010         80.146875         74.53125        1.34375
2011         80.706250         65.68750        1.34375
2012         80.443750         71.09375        1.28125
2013         80.681250         66.65625        1.25000
2014         81.137500         74.40625        1.25000
2015         80.709375         74.87500        1.18750

We see a similar result when it comes to developed countries